sympy/printing/ccode.py

```diff
@@ -219,9 +219,36 @@ def _print_ITE(self, expr):
         return self._print(_piecewise)
 
     def _print_MatrixElement(self, expr):
-        return "{0}[{1}]".format(expr.parent, expr.j +
+        from sympy.matrices import MatrixSymbol, MatrixExpr
+
+        if isinstance(expr.parent, MatrixSymbol):
+            return "{0}[{1}]".format(expr.parent, expr.j +
                 expr.i*expr.parent.shape[1])
 
+        terms = (expr.args[0].args)
+        PREC = precedence(expr.args[0])
+        l = []
+        for term in terms:
+            t = self._print(term)
+            if t.startswith('-'):
```

Comment:
I still believe that this code is in the wrong location. It should be moved to `_print_MatAdd` and re-adapted there. I wouldn't merge this PR with this code in this position.

I also fear this code could potentially introduce bugs, as it might potentially be unable to deal with some special cases of matrix elements.

So, either remove this code from this PR and put it in a new PR, or fix `_print_MatAdd` in this PR.